STAR-Delegation/draft-ietf-acme-star-delegation.md

```diff
@@ -671,6 +671,90 @@ The initial contents of this registry are the extensions defined by the JSON Sch
+against the intended CSR template.  The steps of shaping the CSR template
+correctly, selecting the right CSR template to check against the presented CSR,
+and making sure that the presented CSR matches the selected CSR template are
+all security relevant.
+
+## New ACME Channels
+
+Using the model established in Section 10.1 of {{!RFC8555}}, we can decompose
+the interactions of the basic delegation workflow as shown in
+{{fig-sec-channels}}.
+
+~~~ goat
+{::include art/channels.ascii-art}
+~~~
+{: #fig-sec-channels title="Delegation Channels Topology"}
+
+The considerations regarding the security of the ACME Channel and Validation
+Channel discussed in {{!RFC8555}} apply verbatim to the IdO/ACME server leg.
+The same can be said for the ACME channel on the NDC/IdO leg.  A slightly
+different set of considerations apply to the ACME Channel between NDC and ACME
+server, which consists of a subset of the ACME interface comprising two API
+endpoints: the unauthenticated certificate retrieval and, potentially, non-STAR
+revocation via certificate private key.  No specific security consideration
+apply to the former, but the privacy considerations in Section 6.3 of
+{{!RFC8739}} do.  With regards to the latter, it should be noted that there is
+currently no means for an IdO to disable authorising revocation based on
+certificate private keys.  So, in theory, an NDC could use the revocation API
+directly with the ACME server, therefore by-passing the IdO.  The NDC SHOULD
+NOT directly use the revocation interface exposed by the ACME server unless
+failing to do so would compromise the overall security, for example if the
+certificate private key is compromised and the IdO is not currently reachable.
+
+All other security considerations from {{!RFC8555}} and {{!RFC8739}} apply
+as-is to the delegation topology.
+
+## Limiting the scope of the delegation
+{: #sec-limit-dele }
+
+Any mechanisms an IdO can use to limit the scope of the delegation that seeks
+to be universally available needs to be fully integrated with {{?RFC6125}}
+identity verification.  Unfortunately, {{?RFC6125}} does not define a standard
+way to constrain the identity verification process based on information
+contained in the presented certificate: in fact, the algorithm succeeds if the
+sets of acceptable and presented identifiers have non-empty intersection.  So,
+in an open system there is no standardised way to limit the scope of a
+delegation.
+
+In closed systems, however, a tighter control could be exerted via a locally
+defined {{?RFC6125}} profile.  For example, using a private X.509 extension
+that specifies the allowed IP blocks, the reverse name lookup patterns, or any
+another additional criteria that the service endpoint (i.e., the presenter of
+the delegated certificate) must satisfy in order to be accepted by the client.
+Note that such hypothetical cert extension would be set in the CSR template.
+
```

Comment:
I think we should have a small section about the CNAME. If the CNAME is not validated and just inserted as-is into the zone file, I think this can end up very badly, right?

Reply:
If by "very badly" you mean request redirection doesn't work, yes, there's a risk here.  But that's not different from passing a wrong CNAME on an OOB channel.  I fail to see a security problem though.

Reply:
No, what I had in mind with "CNAME injection", similar to SQL injection - the (malicious) NDC sends a large piece of the DNS zone file that includes the CNAME value but also other resource records that it want to create.

Reply:
That makes total sense.  Thanks for clarifying.

Reply:
We currently have the following text:
```
If an "identifier" object of type "dns" was included, the IdO MUST validate the
specified CNAME at this point in the flow.  The NDC and IdO may have a
pre-established list of valid CNAME values. At the minimum, the IdO MUST verify
that both DNS names are syntactically valid.
```
with an explicit "CNAME validation" requirement.  Do you think we need to add something more to the security considerations?

Reply:
Maybe add to the last sentence, "syntactically valid, to prevent a malicious NDC from injecting arbitrary data into a DNS zone file".

Reply:
OK!

Note: the list of allowed CNAME's is another candidate for the configuration object (#47) alongside the CSR template.